tests/gz-sim-yarp-plugins-check-model/CMakeLists.txt

```
@@ -17,6 +17,16 @@ gsyp_add_tutorial_model_check(forcetorque-model-one-sensor forcetorque/model_one
 gsyp_add_tutorial_model_check(forcetorque-model-two-sensors forcetorque/model_two_sensors/model2sensors.sdf)
 gsyp_add_tutorial_model_check(single-pendulum single_pendulum/model.sdf)
 
+if(ICUB_FOUND)
+    option(ENABLE_COUPLED_PENDULUM_TEST "Enable coupled pendulum tests that require ICUB" ON)
+else()
+    set(ENABLE_COUPLED_PENDULUM_TEST OFF)
+endif()
+
+if(ENABLE_COUPLED_PENDULUM_TEST)
```

```suggestion
if(GZ_SIM_YARP_PLUGINS_ENABLE_TESTS_WITH_ICUB_MAIN)
```

See the other suggestions for the rationale. Furthermore, I would also add `-DGZ_SIM_YARP_PLUGINS_ENABLE_TESTS_WITH_ICUB_MAIN:BOOL=ON` in the tests, so we avoid that for some reason ICUB is not available and the tests silently ignored instead of failing.